tests/system/test_dbapi.py

```diff
@@ -544,6 +570,245 @@ def _insert_row(self, i):
             """
         )
 
+    def test_commit_abort_retry(self, dbapi_database):
+        """Test that when commit failed with Abort exception, then the retry
+        succeeds with transaction having insert as well as query type of
+        statements along with batch dml statements.
+        We are trying to test all types of statements like execute, executemany,
+        fetchone, fetchmany, fetchall"""
+
+        method_count_interceptor = dbapi_database._method_count_interceptor
+        method_count_interceptor.reset()
+        # called 2 times
+        self._insert_row(1)
+        # called 2 times
+        self._cursor.execute("SELECT * FROM contacts")
+        self._cursor.fetchall()
+        self._cursor.execute("start batch dml")
+        self._insert_row(2)
+        self._insert_row(3)
+        # called 2 times for batch dml rpc
+        self._cursor.execute("run batch")
+        row_data = [
+            (4, "first-name4", "last-name4", "test.email4@example.com"),
+            (5, "first-name5", "last-name5", "test.email5@example.com"),
+        ]
+        # called 2 times for batch dml rpc
+        self._cursor.executemany(
+            """
+                INSERT INTO contacts (contact_id, first_name, last_name, email)
+                VALUES (%s, %s, %s, %s)
+            """,
+            row_data,
+        )
+        # called 2 times and as this would make 3 execute streaming sql calls
+        # so total 6 calls
+        self._cursor.executemany(
+            """SELECT * FROM contacts WHERE contact_id = %s""",
+            ((1,), (2,), (3,)),
+        )
+        self._cursor.fetchone()
+        self._cursor.fetchmany(2)
```

Comment:
Do we have a test that shows that this works as expected if the following happens:
1. `executemany` is called with a query and 3 parameter values, and each of the queries returns 1 row (so the same as in this test).
2. `fetchone` is called. That returns the first row.
3. `fetchmany(2)` is called. The first `ExecuteStreamingSql` RPC succeeds. The second `ExecuteStreamingSql` RPC fails with `Aborted`

Reply:
We dont have a test when fetch calls results in Aborted error as I am not able to simulate that like in this case how can we make sure that there are multiple `ExecuteStreamingSql` calls?

Reply:
As discussed created a bug https://buganizer.corp.google.com/issues/319404848 for this and will take it later as part of separate PR

Reply:
ACK